sdk/storage/Azure.Storage.Blobs/AzureStorageNetMigrationV12.md

```diff
@@ -0,0 +1,264 @@
+# Migration Guide: From Microsoft.Azure.Storage.Blob to Azure.Storage.Blobs
+
+This guide intends to assist customers in migrating from the legacy version 11 of the Azure Storage .NET library for Blobs to version 12.
+It will focus on side-by-side comparisons for similar operations between the v12 package, [`Azure.Storage.Blobs`](https://www.nuget.org/packages/Azure.Storage.Blobs) and v11 package, [`Microsoft.Azure.Storage.Blob`](https://www.nuget.org/packages/Microsoft.Azure.Storage.Blob/).
+
+Familiarity with the v11 client library is assumed. For those new to the Azure Storage Blobs client library for .NET, please refer to the [Quickstart](https://docs.microsoft.com/azure/storage/blobs/storage-quickstart-blobs-dotnet) for the v12 library rather than this guide.
+
+## Table of contents
+
+- [Migration benefits](#migration-benefits)
+- [General changes](#general-changes)
+  - [Authentication](#authentication)
+  - [Package and namespaces](#package-and-namespaces)
+  - [Client hierarchy](#client-hierarchy)
+  - [Client constructors](#client-constructors)
+- [Migration samples](#migration-samples)
+  - [Creating a Container](#creating-a-container)
+  - [Uploading Blobs to a Container](#uploading-blobs-to-a-container)
+  - [Downloading Blobs from a Container](#downloading-blobs-from-a-container)
+  - [Listing Blobs in a Container](#listing-blobs-in-a-container)
+  - [Other](#other)
+- [Additional information](#additional-information)
+
+## Migration benefits
+
+To understand why we created our version 12 client libraries, you may refer to the Tech Community blog post, [Announcing the Azure Storage v12 Client Libraries](https://techcommunity.microsoft.com/t5/azure-storage/announcing-the-azure-storage-v12-client-libraries/ba-p/1482394) or refer to our video [Introducing the New Azure SDKs](https://aka.ms/azsdk/intro).
+
+Included are the following:
+- Thread-safe synchronous and asynchronous APIs
+- Improved performance
+- Consistent and idiomatic code organization, naming, and API structure, aligned with a set of common guidelines
+- The learning curve associated with the libraries was reduced
+
+Note: The blog post linked above announces deprecation for previous versions of the library.
+
+## General changes
+
+### Package and namespaces
+
+Package names and the namespaces root for version 12 Azure client libraries follow the pattern `Azure.[Area].[Service]` where the legacy libraries followed the pattern `Microsoft.Azure.[Area].[Service]`.
+
+In this case, to install the legacy v11 package with Nuget:
+```
+dotnet add package Microsoft.Azure.Storage.Blob
+```
+
+It is now the following for v12:
+```
+dotnet add package Azure.Storage.Blobs
+```
+
+### Authentication
+
+#### Managed Identity
```

Comment:
I'm not sure there's a short sample we can provide for this since it's not specific to v11 to v12 and more in the identity libraries. I think a complex stand alone solution/tutorial using MSI might be more helpful (since it requires step by step instructions on how to set up managed identity which is outside the scope of storage.

Reply:
Maybe we should provide a sample for authenticating with AAD (e.g. `TokenCredential`) instead